tooling/lib/releases_diff.rb

```diff
@@ -1,24 +1,44 @@
 require 'yaml'
 
 class ReleasesDiff
-    # Path to the manifest relative to the root of the source tree
+    # Path to the partial manifest relative to the root of the source tree
     attr_accessor :manifest_path
-    # Absolute path to the old manifest
+    # Absolute path to the assembled old manifest
     attr_accessor :old_manifest
-    # Absolute path to the current manifest
+    # Absolute path to the assembled current manifest
     attr_accessor :current_manifest
+    # Absolute path to the temp space
+    attr_accessor :temp_work_dir
 
     def initialize(manifest_path = nil, output = nil)
-        # Path where the manifest lives (inside the source tree)
-        @manifest_path = manifest_path ? manifest_path : "container-host-files/etc/scf/config/role-manifest.yml"
+        # Path where the current partial manifest lives (inside the source tree)
+        current_manifest_path = manifest_path ? manifest_path : "container-host-files/etc/scf/config/role-manifest.yml"
         # The temp work dir used to download older releases
         @temp_work_dir='/tmp/scf-releases-diff'
-        # Path where we save the old manifest
-        @old_manifest="#{@temp_work_dir}/manifest.yaml"
-        # Path to the current manifest
-        @current_manifest=File.join(ReleasesDiff.git_root, @manifest_path)
+        # Path where we save the fully assembled old manifest
+        @old_manifest="#{@temp_work_dir}/old_manifest.yaml"
+        # Path where we save the fully assembled current manifest
+        @current_manifest="#{@temp_work_dir}/current_manifest.yaml"
         # A path to an empty properties file, to be used as opinions
         @empty_opinions_path=File.join(ReleasesDiff.git_root, 'tooling', 'empty_opinions.yaml')
+        # Path where we save the releases of the old manifest
+        @old_releases="#{@temp_work_dir}/old_releases.yaml"
+
+        # Assemble old and current manifests from the pieces. See also `bin/fissile`.
+
+        # Path to the current partial manifest
+        @current_src_path=File.join(ReleasesDiff.git_root, current_manifest_path)
+
+        # Path to the anchors for the current stack
+        if "#{ENV['USE_SLE_BASE']}" == "false"
+            stack = ENV['FISSILE_LIGHT_OPEN42']
+        else
+            stack = ENV['FISSILE_LIGHT_SLE12']
+        end
+
+        system("  cat #{stack}   #{@current_src_path}                     > #{@current_manifest}")
+        system("( cat #{stack} ; git show HEAD:#{current_manifest_path} ) > #{@old_manifest}")
```

Comment:
Using `system` seems a bit icky :(

Reply:
IIRC was the fastest way to hammer it into a working thing without spending time on how to exec subprocesses from ruby, and/or assemble files from pieces.

Better code appreciated

Reply:
I'll try to do that as part of the follow-up bump.